### DISCUSSION TITLE
WebSocket transport not available - Error in run method

### DISCUSSION THREAD

**Discussion by itsomkathe** [miguelgrinberg/Flask-SocketIO#1936] [Q&A]
Hello,

I'm trying to run the [project](https://github.com/bibs2091/Anomaly-detection-system). I'm using docker for this. When it tries to run the `server.py` , I get the following error

![Capture](https://user-images.githubusercontent.com/34516916/216395248-123df29f-22f2-4c92-a3d2-1e89aa6074be.PNG)

While I tried passing the `allow_unsafe_werkzeug=True` into `run`  method, the error still persists. 

I'm using `Flask-SocketIO v5.3.2`

I installed the `simple-websocket`, but still no success. Do I need to use `simple-websocket`  instead of `flask_socketio`? 

Please suggest some solution. Thank you.

**Comment by miguelgrinberg** [comment]
Have you considered accepting the advice the error message is giving you and switching to a production web server? Because this error message is not supposed to be ignored, it is supposed to make you realize that maybe you are making a mistake.

If you insist that you want to use Werkzeug for this use case, then my guess is that the mistake is in how you are running the server, so please share your code.

**Reply by itsomkathe** [reply]
Sure. Here is the [app.js](https://github.com/bibs2091/Anomaly-detection-system/blob/master/model/app.py)

**Reply by miguelgrinberg** [reply]
This code does not have an exception to allow Werkzeug for production use.

Also, you have ignored what I just said. The best solution is to not do this. Use a production web server, not Werkzeug.

**Reply by itsomkathe** [reply]
Okay, I'll try it. But what if I just want to run it just locally. Actually it's for educational purposes.

**Reply by miguelgrinberg** [reply]
So basically you are saying that you are against using a different web server that is more robust than Werkzeug? You can do what you want, I'm just saying that you seem to be jumping through hoops while ignoring the easiest solution.

If you want to keep using Werkzeug you can, I just don't understand why. And why you are showing me code that isn't trying to include the exception that makes Werkzeug work, since that is what you probably did wrong.

**Comment by bluthen** [comment]
From the error message:
Pass allow_unsafe_werkzeug=True to the run() method to disable this error.
